powermon/protocols/pi30.py

```diff
@@ -826,6 +786,48 @@
     },
 }
 
+NEW_FORMAT_COMMANDS = {
+    "QID": {
+        "name": "QID",
+        "description": "Device Serial Number inquiry",
+        "help": " -- queries the device serial number",
+        "result_type": ResultType.SINGLE,
+        "reading_definitions": [{"description": "Serial Number", "reading_type": ReadingType.MESSAGE, "response_type": ResponseType.STRING}],
+        "test_responses": [b"(9293333010501\xbb\x07\r"],
+    },
+    "Q1": {
+        "name": "Q1",
+        "description": "Q1 query",
+        "result_type": ResultType.INDEXED,
```

Comment:
Not sure this is actually indexed, it really is position in result (which was positional in mppsolar before I started to migrate protocols decode.
We should use 'correct' approach from the start since we're changing everything

Reply:
Im thinking this is `ResultType.POSITION` and we has `"position": 0` and index an be kept for results like ved `"response": [ ["V", "Main or channel 1 battery voltage", "V", "float:r/1000"],`